projects/packages/blocks/src/class-blocks.php

```diff
@@ -40,6 +40,22 @@ class Blocks {
 	 * @return WP_Block_Type|false The registered block type on success, or false on failure.
 	 */
 	public static function jetpack_register_block( $slug, $args = array() ) {
+		$block_type = null;
+
+		// If the path to block.json is passed, find the slug in the file then create a block type
+		// object to register the block.
+		// Note: passing the path directly to register_block_type seems to loose the interactivity of
+		// the block once in the editor once it's out of focus.
+		if ( '/' === substr( $slug, 0, 1 ) ) {
+			$metadata = self::get_block_metadata_from_file( $slug );
```

Comment:
Does this happen on build or in runtime? If it's a runtime is it slowing down the registration?

Reply:
At runtime. I assume it would be fast since passing the path to `block.json` when calling [register_block_type](https://developer.wordpress.org/reference/functions/register_block_type/) is the recommended way to register blocks.

Reply:
Core ended up moving away from parsing individual block.json files: https://make.wordpress.org/core/2022/10/07/improved-php-performance-for-core-blocks-registration/